pkg/apis/config/defaults.go

```diff
@@ -71,24 +71,30 @@ func NewDefaultsConfigFromConfigMap(config *corev1.ConfigMap) (*Defaults, error)
 // Defaults includes the default values to be populated by the webhook.
 type Defaults struct {
 	// NamespaceDefaultsConfig are the default Broker Configs for each namespace.
-	// Namespace is the key, the value is the KReference to the config.
-	NamespaceDefaultsConfig map[string]*ClassAndBrokerConfig `json:"namespaceDefaults,omitempty"`
+	// Namespace is the key, the value is the NamespaceDefaultConfig
+	NamespaceDefaultsConfig map[string]*DefaultConfig `json:"namespaceDefaults,omitempty"`
 
-	// ClusterDefaultBrokerConfig is the default broker config for all the namespaces that
-	// are not in NamespaceDefaultBrokerConfigs.
-	ClusterDefault *ClassAndBrokerConfig `json:"clusterDefault,omitempty"`
+	// ClusterDefaultConfig is the default broker config for all the namespaces that
+	// are not in NamespaceDefaultBrokerConfigs. Different broker class could have
+	// different default config.
+	ClusterDefaultConfig *DefaultConfig `json:"clusterDefault,omitempty"`
 }
 
-// ClassAndBrokerConfig contains configuration for a given namespace for broker. Allows
-// configuring the Class of the Broker, the reference to the
-// config it should use and it's delivery.
-type ClassAndBrokerConfig struct {
-	BrokerClass   string `json:"brokerClass,omitempty"`
-	*BrokerConfig `json:",inline"`
+// DefaultConfig struct contains the default configuration for the cluster and namespace.
+type DefaultConfig struct {
+	// DefaultBrokerClass and DefaultBrokerClassConfig are the default broker class for the whole cluster
```

Comment:
I only see `DefaultBrokerClass` in the struct

Reply:
*BrokerConfig      `json:",inline"`
Should be the config. The comment has been updated accordingly.

Reply:
https://github.com/knative/eventing/pull/7631/files/63954533b2b81af9ce89b0e8a2c17a08bd1c1111#diff-730a26cf5aea2aaa8dd90f1e5f8eba349fedca231d776a5253d34d67e9c2b3bdR87-R88

The change is here